metadata/src/main/java/org/apache/kafka/image/publisher/SnapshotGenerator.java

```diff
@@ -200,7 +201,24 @@ void resetSnapshotCounters() {
     }
 
     @Override
-    public void publishSnapshot(
+    public void onMetadataUpdate(
+        MetadataDelta delta,
+        MetadataImage newImage,
+        LoaderManifest manifest
+    ) {
+        switch (manifest.type()) {
+            case LOG_DELTA:
+                publishLogDelta(delta, newImage, (LogDeltaManifest) manifest);
+                break;
+            case SNAPSHOT:
+                publishSnapshot(delta, newImage, (SnapshotManifest) manifest);
+                break;
+            default:
+                break;
```

Comment:
wdyt about throwing an IllegalStateException in the `default`. I do that sometimes to future-proof against unhandled enums down the road.

Reply:
Actually, let's just remove the `default` case. spotbugs will flag non-exhaustive switch statements on enums. And that's a hard build error for us.